src/t_zset.c

```diff
@@ -3474,7 +3418,7 @@ void genericZrangebylexCommand(zrange_result_handler *handler,
 
             /* We know the element exists, so ziplistGet should always
              * succeed. */
-            serverAssertWithInfo(c,zobj,ziplistGet(eptr,&vstr,&vlen,&vlong));
```

Comment:
`ziplistGet` cound returns 0, but `lpGet` never does, so these assertions aren't specific to `vstr`, so I removed them.

Reply:
so it looks like there are plenty of other assertions to remove.
in each of the 5 places that i commented, there was a similar line just above it that still has that assertion

Reply:
I performed an inspection.
Those assertions are required, they are used to avoid corrupt data.

Reply:
You mean that some are still required, and others are not?

Reply:
Yeah.

Reply:
ok, got it.
in `ziplistGet` the return of 0 is in case `p` is NULL, or points to `ZIP_END`.
for `ziplistGet` there's an internal assertions for `p`, and the `LP_EOF` is already checked in `lpNext`.
the other assertions i saw you didn't remove are in calls to `lpNext` (i didn't notice).